### PR TITLE
Add support in vbus component for Deltasol BS 2009

### PR DESCRIPTION
**Pull request by clydebarrow**
# What does this implement/fix?

This extends the `vbus` component by adding support for the Deltasol BS 2009 and Dux H3214 solar controllers.

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other

**Related issue or feature (if applicable):** fixes <link to issue>

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):** esphome/esphome-docs#2992

## Test Environment

- [x] ESP32
- [ ] ESP32 IDF
- [ ] ESP8266
- [ ] RP2040

## Example entry for `config.yaml`:

```yaml
# Example config.yaml excerpt
uart:
  id: resol
  rx_pin: GPIO10
  baud_rate: 9600

vbus:
  uart_id: resol

sensor:
  - platform: vbus
    model: deltasol_bs_2009
    temperature_1:
      name: Collector Temp
```

## Checklist:
  - [x] The code change is tested and works locally.
  - [ ] Tests have been added to verify that the new code works (under `tests/` folder).

If user exposed functionality or configuration variables are added/changed:
  - [x] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).

### CODE REVIEW

File: esphome/components/vbus/binary_sensor/vbus_binary_sensor.h [hunk 1]
```diff
@@ -39,6 +39,36 @@ class DeltaSolBSPlusBSensor : public VBusListener, public Component {
...
+  binary_sensor::BinarySensor *s4_error_bsensor_{nullptr};
+  binary_sensor::BinarySensor *collector_max_bsensor_{nullptr};
+  binary_sensor::BinarySensor *collector_min_bsensor_{nullptr};
+  binary_sensor::BinarySensor *collector_frost_bsensor_{nullptr};
+  binary_sensor::BinarySensor *tube_collector_bsensor_{nullptr};
+  binary_sensor::BinarySensor *recooling_bsensor_{nullptr};
+  binary_sensor::BinarySensor *hqm_bsensor_{nullptr};
+
+  void handle_message(std::vector<uint8_t> &message) override;
+};
```
Comment (ssieb) [inline esphome/components/vbus/binary_sensor/vbus_binary_sensor.h:61]:
```suggestion
};

```

**Comment by ssieb** [comment]
You're missing the `to_code` section for the binary sensors.

**Comment by ssieb** [comment]
Ignore the CI failures for now.  There's something wrong today.

**Comment by clydebarrow** [comment]
@ssieb I think this is all done now. The CI is still failing, but I've run the checks locally and fixed the reported formatting issues.

I did not see any unit tests for the vbus component so did not add any test code. Point me in the right direction if this is required.

**Review by ssieb — APPROVED** [review]

**Pull request merged**